### PR TITLE
Mark VertexAI AutoMLText deprecation

### PR DESCRIPTION
**Pull request by olegkachur-e**
- Label deprecated content
- Update docs

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by MaksYermak — APPROVED** [review]

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py [hunk 1]
```diff
@@ -185,6 +186,11 @@ def get_auto_ml_image_training_job(
             model_encryption_spec_key_name=model_encryption_spec_key_name,
         )
 
+    @deprecated(
+        planned_removal_date="December 15, 2025",
```
Comment (moiseenkov) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py:190]: Should it be here June 15, 2025?

Comment (olegkachur-e) [reply airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py:190]: The idea was June 15 2025(deprecation date) + 6 month.
 Is it incorrect suggestion?

Comment (olegkachur-e) [reply airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]: All deprecation dates got updated to the dates suggested in google documentation, thank you for highlighting this detail!

**Review by moiseenkov — APPROVED** [review]

**Comment by olegkachur-e** [comment]
Thanks for high quality review, closing to re-open in community.

**Pull request closed without merging**